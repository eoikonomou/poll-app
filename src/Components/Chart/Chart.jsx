import React from 'react';
import PropTypes from 'prop-types';
import {
  Bar,
  BarChart,
  CartesianGrid,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';

const Chart = ({
  height,
  width,
  data,
  xDataKey,
  yDataKey,
}) => (
  <BarChart
    height={height}
    width={width}
    data={data}
  >
    <CartesianGrid strokeDasharray="3 3" />
    <XAxis dataKey={xDataKey} />
    <YAxis dataKey={yDataKey} allowDecimals={false} />
    <Tooltip />
    <Bar dataKey={yDataKey} fill="#8884d8" />
  </BarChart>
);

Chart.displayName = 'Chart';

Chart.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  xDataKey: PropTypes.string,
  yDataKey: PropTypes.string,
};

Chart.defaultProps = {
  xDataKey: 'name',
  yDataKey: 'value',
};

export default Chart;
