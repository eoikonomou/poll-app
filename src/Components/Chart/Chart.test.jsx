import React from 'react';
import { shallow } from 'enzyme';
import Chart from './Chart';

describe('Chart', () => {
  test('should render successfully', () => {
    const wrapper = shallow(<Chart height={450} width={480} data={[]} />);
    expect(wrapper.prop('height')).toEqual(450);
    expect(wrapper.prop('width')).toEqual(480);
    expect(wrapper.prop('data')).toEqual([]);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with default properties', () => {
    const wrapper = shallow(
      <Chart
        height={450}
        width={480}
        data={[]}
      />,
    );
    expect(wrapper.find('XAxis').prop('dataKey')).toEqual('name'); // xDataKey
    expect(wrapper.find('YAxis').prop('dataKey')).toEqual('value'); // yDataKey
  });
});
