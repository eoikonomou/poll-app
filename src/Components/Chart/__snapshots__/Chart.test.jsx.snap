// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should render successfully 1`] = `
<BarChart
  barCategoryGap="10%"
  barGap={4}
  data={Array []}
  height={450}
  layout="horizontal"
  margin={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  reverseStackOrder={false}
  stackOffset="none"
  width={480}
>
  <CartesianGrid
    fill="none"
    horizontal={true}
    horizontalFill={Array []}
    horizontalPoints={Array []}
    stroke="#ccc"
    strokeDasharray="3 3"
    vertical={true}
    verticalFill={Array []}
    verticalPoints={Array []}
  />
  <XAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    dataKey="name"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={30}
    hide={false}
    mirror={false}
    orientation="bottom"
    padding={
      Object {
        "left": 0,
        "right": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="category"
    width={0}
    xAxisId={0}
  />
  <YAxis
    allowDataOverflow={false}
    allowDecimals={false}
    allowDuplicatedCategory={true}
    dataKey="value"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={0}
    hide={false}
    mirror={false}
    orientation="left"
    padding={
      Object {
        "bottom": 0,
        "top": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="number"
    width={60}
    yAxisId={0}
  />
  <Tooltip
    active={false}
    allowEscapeViewBox={
      Object {
        "x": false,
        "y": false,
      }
    }
    animationDuration={400}
    animationEasing="ease"
    contentStyle={Object {}}
    coordinate={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    cursor={true}
    cursorStyle={Object {}}
    filterNull={true}
    isAnimationActive={true}
    itemStyle={Object {}}
    labelStyle={Object {}}
    offset={10}
    separator=" : "
    useTranslate3d={false}
    viewBox={
      Object {
        "x1": 0,
        "x2": 0,
        "y1": 0,
        "y2": 0,
      }
    }
    wrapperStyle={Object {}}
  />
  <Bar
    animationBegin={0}
    animationDuration={400}
    animationEasing="ease"
    data={Array []}
    dataKey="value"
    fill="#8884d8"
    hide={false}
    isAnimationActive={true}
    layout="vertical"
    legendType="rect"
    minPointSize={0}
    onAnimationEnd={[Function]}
    onAnimationStart={[Function]}
    xAxisId={0}
    yAxisId={0}
  />
</BarChart>
`;
